### PR TITLE
Add interpolation to renderer.

### PR DESCRIPTION
**Pull request by siarheishut**
Add interpolation logic to MapRenderer and fix tests according to it.

### CODE REVIEW

File: src/map_renderer.cpp [hunk 1]
```diff
@@ -68,10 +72,16 @@ CombineCoords(
 
 std::unordered_map<std::string_view, svg::Point> TransformCoords(
     const rm::renderer_utils::Buses &buses,
-    const rm::renderer_utils::Stops &stops,
+    rm::renderer_utils::Stops &stops,
     const rm::renderer_utils::Frame &frame) {
   using namespace rm::coords_converter;
 
+  std::unordered_set < std::string_view > base_stops;
```
Comment (foxfrikses) [inline src/map_renderer.cpp:79]: Using `BaseStops`

File: src/map_renderer.cpp [hunk 2]
```diff
@@ -68,10 +72,16 @@ CombineCoords(
...
+  rm::Combine(base_stops, EndPoints(buses), IntersectionsCrossRoute(buses),
```
Comment (foxfrikses) [inline src/map_renderer.cpp:80]: I would put each on it's own line for better readability:
```
rm::Combine(base_stops, 
            EndPoints(buses), 
            IntersectionsCrossRoute(buses),
            IntersectionsWithinRoute(buses, 3));
```

File: src/map_renderer.cpp [hunk 3]
```diff
@@ -68,10 +72,16 @@ CombineCoords(
...
+              IntersectionsWithinRoute(buses, 3));
+  for (auto &bus : buses) {
+    stops = Interpolate(std::move(stops), bus.second.route, base_stops);
```
Comment (foxfrikses) [inline src/map_renderer.cpp:83]: I think you need to provide an expanded route here.

Comment (siarheishut) [reply src/map_renderer.cpp:83]: Done.

File: src/map_renderer.cpp (hunk 1 above)
Comment (siarheishut) [reply src/map_renderer.cpp:79]: Done.

File: src/map_renderer.cpp [hunk 4]
```diff
@@ -68,10 +72,19 @@ CombineCoords(
...
     const rm::renderer_utils::Frame &frame) {
   using namespace rm::coords_converter;
 
+  BaseStops base_stops;
+  rm::Combine(base_stops,
+              EndPoints(buses),
+              IntersectionsCrossRoute(buses),
+              IntersectionsWithinRoute(buses, 3));
+  for (auto &bus : buses) {
+    auto expanded_route = rm::renderer_utils::ExpandRoute(bus.second.route);
```
Comment (foxfrikses) [inline src/map_renderer.cpp:85]: Why are you expanding all the types of trips?

Comment (foxfrikses) [reply src/map_renderer.cpp:85]: Do you by the way even need to expand anything?

Like, you have a roundtrip — nothing to expand.
Non-roundtrip — expand way A-b-c-d-E to A-b-c-d-E-d-c-b-A. A and E are always base, as you've put all the endpoints into base.  And the path between A and E will just be repeated twice. And as there is no difference in computing the Interpolation in b-c-d and in d-c-b, I don't think you even need to expand anything, an I right?

I remember you had expansion logic in first versions of the Interpolate function. But I cannot remember why you needed it. Did you really need it?

Comment (siarheishut) [reply src/map_renderer.cpp:85]: In fact I don't need it here. I was thinking about it earlier but then refactoring started and I forgot to change it in a hurry. Removed.

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**